feat: mastra mcp agent (#1505)
apify/apify-docs: sources/platform/integrations/ai/mastra.md

---
title: Mastra MCP integration
sidebar_label: Mastra
description: Learn how to build AI Agents with Mastra via Apify Actors MCP server
sidebar_position: 1
slug: /integrations/mastra
---

> **Learn how to build AI Agents with Mastra and Apify Actors MCP Server.**

Reviewer: As per [here](https://www.notion.so/apify/Apify-writing-style-guide-de9fbb99dcd84665b6d3f790fc88b3b6?pvs=4#1c0f39950a2280679b50cf187add97ee)

```suggestion
**Learn how to build AI agents with Mastra and Apify Actors MCP Server.**
```

---

## What is Mastra

[Mastra](https://mastra.ai) is an open-source TypeScript framework for building AI applications efficiently. It provides essential tools like agents, workflows, retrieval-augmented generation (RAG), integrations, and evaluations. Supporting any LLM (e.g., GPT-4, Claude, Gemini). You can run it locally or deploy it to a serverless cloud like [Apify](https://apify.com).

:::note Explore Mastra

> See the [Mastra docs](https://mastra.ai/docs) for more.

Reviewer: We try to avoid using language like see for accessibility reasons, also more what? This feels unfinished. Perhaps

```suggestion
Check out the [Mastra docs](https://mastra.ai/docs) for more information.
```

## How to use Apify with Mastra via MCP

This guide demonstrates how to integrate Apify Actors with Mastra by building an agent that uses the [RAG Web Browser](https://apify.com/apify/rag-web-browser) Actor to search Google for TikTok profiles and the [TikTok Data Extractor](https://apify.com/clockworks/free-tiktok-scraper) Actor to extract and analyze data from the TikTok profiles via MCP.

### Prerequisites

- _Apify API token_: To use Apify Actors, you need an Apify API token. Learn how to obtain it in the [Apify documentation](https://docs.apify.com/platform/integrations/api).

> - _OpenAI API key_: To power the agents, you need an OpenAI API key. Get one from the [OpenAI platform](https://platform.openai.com/account/api-keys).

Reviewer: Perhaps here instead of explicitly mentioning OpenAI we cold say "LLM provider API key", and leave example as is? At most add comments within code on line where we mention OpenAI that you can use any other as well.

> - **Apify API token**: To use Apify Actors, you need an Apify API token. Learn how to obtain it in the [Apify documentation](https://docs.apify.com/platform/integrations/api).
> - **OpenAI API key**: To power the agents, you need an OpenAI API key. Get one from the [OpenAI platform](https://platform.openai.com/account/api-keys).
> - **Node.js**: Ensure you have Node.js installed.
> - **Packages**: Install the following packages:

Reviewer: Bold text is reserved to highlight UI elements, in this instance I would suggest using emphasis.

```suggestion
- _Apify API token_: To use Apify Actors, you need an Apify API token. Learn how to obtain it in the [Apify documentation](https://docs.apify.com/platform/integrations/api).
- _OpenAI API key_: To power the agents, you need an OpenAI API key. Get one from the [OpenAI platform](https://platform.openai.com/account/api-keys).
- _Node.js_: Ensure you have Node.js installed.
- _Packages_: Install the following packages:
```
